File: javascript/space-age/space-age.js
//@ts-check

const YEAR_PLANET_CONVERSION = {
  'mercury': 0.2408467,
  'venus': 0.61519726,
  'mars': 1.8808158,
  'earth': 1,
  'jupiter': 11.862615,
  'saturn': 29.447498,
  'uranus': 84.016846,
  'neptune': 164.79132
};

const SECONDS_IN_YEAR_ON_EARTH = 31557600;

/**
 * Return age in years on different planets
 * @param {string} planet name
 * @param {number} timeSec, Earth age in seconds
 * return {number} age in years
 */
export const age = (planet, timeSec) => 
  Number((timeSec / SECONDS_IN_YEAR_ON_EARTH 
    / YEAR_PLANET_CONVERSION[planet]).toFixed(2))

File: c/space-age/space_age.c
#include "space_age.h"

#define SECONDS_PER_YEAR_ON_EARTH (365.25 * 24 * 60 * 60)

struct Planet planets[] = {
  {MERCURY, 0.2408467},
  {VENUS, 0.61519726},
  {EARTH, 1.0},
  {MARS, 1.8808158},
  {JUPITER, 11.862615},
  {SATURN, 29.447498},
  {URANUS, 84.016846},
  {NEPTUNE, 164.79132}
};


float age(planet_t planet, int64_t seconds) {
  return (seconds / SECONDS_PER_YEAR_ON_EARTH / planets[planet].orbitalPeriod);
}
